************* Module C3_W2_Assignment_Solution
python-scripts\C3_W2_Assignment_Solution.py:1:0: C0114: Missing module docstring (missing-module-docstring)
python-scripts\C3_W2_Assignment_Solution.py:1:0: C0103: Module name "C3_W2_Assignment_Solution" doesn't conform to snake_case naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:98:0: C0103: Constant name "dirname" doesn't conform to UPPER_CASE naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:101:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
python-scripts\C3_W2_Assignment_Solution.py:114:0: C0103: Constant name "n_lines" doesn't conform to UPPER_CASE naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:180:25: C0103: Argument name "EOS_int" doesn't conform to snake_case naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:180:19: W0621: Redefining name 'line' from outer scope (line 102) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:195:8: C0103: Variable name "c" doesn't conform to snake_case naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:265:0: R0914: Too many local variables (19/15) (too-many-locals)
python-scripts\C3_W2_Assignment_Solution.py:265:19: W0621: Redefining name 'batch_size' from outer scope (line 533) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:265:31: W0621: Redefining name 'max_length' from outer scope (line 534) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:265:55: W0621: Redefining name 'line_to_tensor' from outer scope (line 180) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:311:8: W0621: Redefining name 'line' from outer scope (line 102) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:324:12: W0621: Redefining name 'batch' from outer scope (line 742) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:328:16: C0103: Variable name "li" doesn't conform to snake_case naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:382:0: W0104: Statement seems to have no effect (pointless-statement)
python-scripts\C3_W2_Assignment_Solution.py:413:0: C0413: Import "import itertools" should be placed at the top of the module (wrong-import-position)
python-scripts\C3_W2_Assignment_Solution.py:474:0: C0103: Function name "GRULM" doesn't conform to snake_case naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:487:4: W0621: Redefining name 'model' from outer scope (line 502) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:533:0: C0103: Constant name "batch_size" doesn't conform to UPPER_CASE naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:534:0: C0103: Constant name "max_length" doesn't conform to UPPER_CASE naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:547:17: W0621: Redefining name 'lines' from outer scope (line 99) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:547:24: W0621: Redefining name 'max_length' from outer scope (line 534) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:557:4: W0621: Redefining name 'n_lines' from outer scope (line 114) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:558:8: C0103: Variable name "l" doesn't conform to snake_case naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:563:0: C0103: Constant name "num_used_lines" doesn't conform to UPPER_CASE naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:566:0: C0103: Constant name "steps_per_epoch" doesn't conform to UPPER_CASE naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:609:0: C0413: Import "from trax.supervised import training" should be placed at the top of the module (wrong-import-position)
python-scripts\C3_W2_Assignment_Solution.py:613:0: W0102: Dangerous default value lines (builtins.list) as argument (dangerous-default-value)
python-scripts\C3_W2_Assignment_Solution.py:613:0: W0102: Dangerous default value eval_lines (builtins.list) as argument (dangerous-default-value)
python-scripts\C3_W2_Assignment_Solution.py:613:0: R0913: Too many arguments (8/5) (too-many-arguments)
python-scripts\C3_W2_Assignment_Solution.py:613:16: W0621: Redefining name 'model' from outer scope (line 502) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:613:23: W0621: Redefining name 'data_generator' from outer scope (line 265) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:613:39: W0621: Redefining name 'batch_size' from outer scope (line 533) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:613:54: W0621: Redefining name 'max_length' from outer scope (line 534) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:613:69: W0621: Redefining name 'lines' from outer scope (line 99) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:613:82: W0621: Redefining name 'eval_lines' from outer scope (line 136) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:649:4: W0621: Redefining name 'training_loop' from outer scope (line 666) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:714:15: W0621: Redefining name 'preds' from outer scope (line 743) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:730:4: W0621: Redefining name 'log_ppx' from outer scope (line 744) (redefined-outer-name)
python-scripts\C3_W2_Assignment_Solution.py:768:4: C0103: Variable name "u" doesn't conform to snake_case naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:769:4: C0103: Variable name "g" doesn't conform to snake_case naming style (invalid-name)
python-scripts\C3_W2_Assignment_Solution.py:772:0: C0116: Missing function or method docstring (missing-function-docstring)
python-scripts\C3_W2_Assignment_Solution.py:774:13: R1721: Unnecessary use of a comprehension, use list(prefix) instead. (unnecessary-comprehension)
python-scripts\C3_W2_Assignment_Solution.py:69:0: C0411: standard import "import pickle" should be placed before "import trax" (wrong-import-order)
python-scripts\C3_W2_Assignment_Solution.py:71:0: C0411: standard import "import random as rnd" should be placed before "import trax" (wrong-import-order)
python-scripts\C3_W2_Assignment_Solution.py:413:0: C0411: standard import "import itertools" should be placed before "import trax" (wrong-import-order)
python-scripts\C3_W2_Assignment_Solution.py:69:0: W0611: Unused import pickle (unused-import)
python-scripts\C3_W2_Assignment_Solution.py:72:0: W0611: Unused fastmath imported from trax (unused-import)

------------------------------------------------------------------
Your code has been rated at 6.73/10 (previous run: 6.73/10, +0.00)

